src/stk/databases/mongo_db/molecule.py

```diff
@@ -296,20 +296,27 @@ def put(self, molecule):
         return self._put(HashableDict(json))
 
     def _put(self, json):
-        # insert_one() corrupts the state of the dict it is passed
-        # as an argument (it adds various items to it).
-        # Using insert_one(json['molecule']) would mean that the json
-        # in the lru_cache is modified with some extra items added by
-        # insert_one(). This means that the next time _put() is used
-        # with a clean json, it will not match the one in the cache,
-        # because the one in the cache has the extra items added by
-        # insert_one(). To prevent this use
-        # insert_one(dict(json['molecule'])), which means that a copy
-        # is modified by insert_one and the json in the cache is
-        # not changed.
-
-        self._molecules.insert_one(dict(json['molecule']))
-        self._position_matrices.insert_one(dict(json['matrix']))
+        keys = dict(json['matrix'])
+        keys.pop('m')
+
+        query = {'$or': []}
+        for key, value in keys.items():
+            query['$or'].append({key: value})
+
+        self._molecules.update_many(
```

Comment:
I assume that both the `molecules` and `position_matrices` databases are using identical molecule keys?

Reply:
Correct! That's how you are able to reconstruct your molecule when you supply a given key.